pkg/hostap/monitor.go

```diff
@@ -240,36 +261,56 @@ func (m *InterfaceMonitor) handleIfEvents() {
 			close(m.ifEventCh)
 			return
 		case linkUpdateEvent := <-m.ifEventCh:
-			newOpState := linkUpdateEvent.Link.Attrs().OperState
-			if newOpState == m.operState {
-				level.Info(m.Logger).Log("interface", "event", m.IfName, "op state not changed", m.operState)
-				continue
+			if linkUpdateEvent.Link.Attrs().OperState != m.operState {
+				m.handlePfEventForOpStateChange(linkUpdateEvent)
+			} else {
+				m.handlePfEventForVfVlanChange(linkUpdateEvent)
 			}
-			m.operState = newOpState
-			level.Info(m.Logger).Log("interface", "event", m.IfName, "op state changed", m.operState)
-			m.addrMutex.Lock()
-			if newOpState == netlink.OperDown {
-				for addr := range m.authenticatedAddrs {
-					err := m.deauthenticate(addr)
-					if err != nil {
-						level.Error(m.Logger).Log("interface", "addr", m.IfName, addr, "error deauthenticating addr", err)
-					}
-					m.deauthRequests[addr] = getCurrentTimestamp()
-				}
-				m.logEvent(kapi.EventTypeWarning, "interface is down")
-			} else if newOpState == netlink.OperUp {
-				for addr := range m.authenticatedAddrs {
-					delete(m.deauthRequests, addr)
-				}
-				m.logEvent(kapi.EventTypeNormal, "interface is up")
+		}
+	}
+}
+
+func (m *InterfaceMonitor) handlePfEventForVfVlanChange(linkUpdateEvent netlink.LinkUpdate) {
+	if len(linkUpdateEvent.Link.Attrs().Vfs) == 0 {
+		return
+	}
+	m.addrMutex.Lock()
+	defer m.addrMutex.Unlock()
+	for _, vf := range linkUpdateEvent.Link.Attrs().Vfs {
+		if vfInfo, ok := m.PfInfo.VFs[vf.ID]; ok && vfInfo.Vlan != vf.Vlan {
+			vfInfo.Vlan = vf.Vlan
```

Comment:
only if `vlan != 4095`?

Reply:
good catch ! yes, it avoids an unnecessary netlink call.